README.md

```diff
@@ -77,52 +77,51 @@ If you have a GPU installed on your system, then you can do training.
 
 Run the following commands from the root directory of the repository.
 
-1. Download and prepare the dataset. This requires over a GB of disk space.
+1. Install development requirements.
 ```
-./build/prepare-wmt14 --small
+pip install -r dev-requirements.txt
 ```
-To prepare the full dataset, omit the `--small`. Preparation for the full dataset can take 15GB of disk space and 10-15 minutes on a fast computer. 
 
-2. Activate the virtual environment if it is not already activated.
+2. Download and prepare the dataset. This requires over a GB of disk space.
 ```
-source mt-fast-training/bin/activate
-source .env
+./build/prepare-wmt14 --small
 ```
+To prepare the full dataset, omit the `--small`. Preparation for the full dataset can take 15GB of disk space and 10-15 minutes on a fast computer. 
 
-3. (Optional) View the model architecture.
+3. Activate the virtual environment if it is not already activated.
 ```
-python fr2en.py summary config/densenet-l12.yaml
+source fast-training/bin/activate
+source .env
 ```
 
-4. (Optional) Search for a good learning rate.
+4. (Optional) View the model architecture.
 ```
-streamlit run fr2en.py find_lr config/densenet-l12.yaml
+python fr2en.py summary config/densenet-12.yaml
 ```
-Then go in the browser to http://localhost:8501 where a plot of the learning rate performance will appear. Choose the learning rate where the graph has the steepest decline. Then back in the console, press CTRL-C to exit.
 
-5. Train a model. For example, choose the 12 layer Densenet (`config/densenet-l12.yaml`). This can take a long time.
+5. Train a model. For example, choose the 12 layer Densenet (`config/densenet-12.yaml`). This can take a long time.
 ```
-python fr2en.py train --lr 0.0015 config/densenet-l12.yaml
+python fr2en.py train --lr 0.0015 config/densenet-12.yaml
 ```
 To do distributed training on nVidia GPUs `0` and `1`, run
 ```
-python fr2en.py train --gpu_ids '[0,1]' config/densenet-l12.yaml
+python fr2en.py train --device_ids '[0,1]' config/densenet-12.yaml
```

Comment:
This runs, but I also get this message:

>> python fr2en.py train --device_ids '[0,1]' config/densenet-12.yaml
epoch     train_loss  valid_loss  bleu      time
epoch     train_loss  valid_loss  bleu      time    -------------------------| 0.00% [0/1000 00:00<00:00]
tuple appears in op that does not forward tuples (VisitNode at /pytorch/torch/csrc/jit/passes/lower_tuples.cpp:117)
frame #0: std::function<std::string ()>::operator()() const + 0x11 (0x7f7f38449441 in /mnt/fs03/home/nlopatina/fast-training/fast-training/lib/python3.6/site-packages/torch/lib/libc10.so)
frame #1: c10::Error::Error(c10::SourceLocation, std::string const&) + 0x2a (0x7f7f38448d7a in /mnt/fs03/home/nlopatina/fast-training/fast-training/lib/python3.6/site-packages/torch/lib/libc10.so)
frame #2: <unknown function> + 0xaf61f5 (0x7f7f0ceb31f5 in /mnt/fs03/home/nlopatina/fast-training/fast-training/lib/python3.6/site-packages/torch/lib/libtorch.so.1)
frame #3: <unknown function> + 0xaf6464 (0x7f7f0ceb3464 in /mnt/fs03/home/nlopatina/fast-training/fast-training/lib/python3.6/site-packages/torch/lib/libtorch.so.1)
frame #4: torch::jit::LowerAllTuples(std::shared_ptr<torch::jit::Graph>&) + 0x13 (0x7f7f0ceb34a3 in /mnt/fs03/home/nlopatina/fast-training/fast-training/lib/python3.6/site-packages/torch/lib/libtorch.so.1)
frame #5: <unknown function> + 0x3f84b4 (0x7f7f329e64b4 in /mnt/fs03/home/nlopatina/fast-training/fast-training/lib/python3.6/site-packages/torch/lib/libtorch_python.so)
frame #6: <unknown function> + 0x130cfc (0x7f7f3271ecfc in /mnt/fs03/home/nlopatina/fast-training/fast-training/lib/python3.6/site-packages/torch/lib/libtorch_python.so)
<omitting python frames>
frame #45: <unknown function> + 0x76ba (0x7f7f418fc6ba in /lib/x86_64-linux-gnu/libpthread.so.0)
frame #46: clone + 0x6d (0x7f7f4163241d in /lib/x86_64-linux-gnu/libc.so.6)

tuple appears in op that does not forward tuples (VisitNode at /pytorch/torch/csrc/jit/passes/lower_tuples.cpp:117)
frame #0: std::function<std::string ()>::operator()() const + 0x11 (0x7f48908d1441 in /mnt/fs03/home/nlopatina/fast-training/fast-training/lib/python3.6/site-packages/torch/lib/libc10.so)
frame #1: c10::Error::Error(c10::SourceLocation, std::string const&) + 0x2a (0x7f48908d0d7a in /mnt/fs03/home/nlopatina/fast-training/fast-training/lib/python3.6/site-packages/torch/lib/libc10.so)
frame #2: <unknown function> + 0xaf61f5 (0x7f488f8ca1f5 in /mnt/fs03/home/nlopatina/fast-training/fast-training/lib/python3.6/site-packages/torch/lib/libtorch.so.1)
frame #3: <unknown function> + 0xaf6464 (0x7f488f8ca464 in /mnt/fs03/home/nlopatina/fast-training/fast-training/lib/python3.6/site-packages/torch/lib/libtorch.so.1)
frame #4: torch::jit::LowerAllTuples(std::shared_ptr<torch::jit::Graph>&) + 0x13 (0x7f488f8ca4a3 in /mnt/fs03/home/nlopatina/fast-training/fast-training/lib/python3.6/site-packages/torch/lib/libtorch.so.1)
frame #5: <unknown function> + 0x3f84b4 (0x7f489b4bd4b4 in /mnt/fs03/home/nlopatina/fast-training/fast-training/lib/python3.6/site-packages/torch/lib/libtorch_python.so)
frame #6: <unknown function> + 0x130cfc (0x7f489b1f5cfc in /mnt/fs03/home/nlopatina/fast-training/fast-training/lib/python3.6/site-packages/torch/lib/libtorch_python.so)
<omitting python frames>
frame #45: <unknown function> + 0x76ba (0x7f48a4fdf6ba in /lib/x86_64-linux-gnu/libpthread.so.0)
frame #46: clone + 0x6d (0x7f48a4d1541d in /lib/x86_64-linux-gnu/libc.so.6)

Reply:
I get that message too, and I could not determine the cause. It doesn't seem to cause any problems, so since it is hard to debug, I don't think it should be a high priority.

I added a warning message that the error might show up but doesn't cause any problems.